README.md

```diff
@@ -2,7 +2,74 @@
 
 # Radix tree
 
-Read more about the use cases in the package's docs.
+Radix tree is defined in Wikipedia: <https://en.wikipedia.org/wiki/Radix_tree>
+
+A data structure that represents a space-optimized trie (prefix tree) in which
+each node that is the only child is merged with its parent.
+
+Under the hood, the Tree structure is managed with
+[elm-rosetree](https://package.elm-lang.org/packages/zwilias/elm-rosetree/latest/)
+
+In this package we have two kinds of implementations, one is ordered and the
+other is un-ordered. It's possible that the un-ordered isn't called "Radix tree"
+but the author of the package doesn't know of a better name.
+
+It would be easier to follow by example, or try it on a helpful app like
+this one: <https://www.cs.usfca.edu/~galles/visualization/RadixTree.html>
+
+## Ordered Radix Tree
+
+Say we have the following list of chars:
+
+    [ [ "r", "o", "m", "a", "n", "e" ]
```

Comment:
In the README, I recommend to wrap Elm code blocks inside \```elm```
because then it will also look good on GitHub (I mean it should gain the proper syntax highlighting)